ecosystem/indexer-grpc/indexer-test-transactions/build.rs

```diff
@@ -6,25 +6,74 @@ use std::{env, fs, path::Path};
 
 fn main() {
     let out_dir = env::var("OUT_DIR").unwrap();
+    println!("OUT_DIR: {:?}", out_dir);
     let dest_path = Path::new(&out_dir).join("generate_transactions.rs");
+
+    let mut all_transactions_code = String::new();
+
+    // Create directories if they do not exist
+    create_directory_if_missing("json_transactions/imported_mainnet_txns");
+    create_directory_if_missing("json_transactions/imported_testnet_txns");
+    create_directory_if_missing("json_transactions/generated_txns");
+
+    // Process each directory and generate code for it
+    all_transactions_code.push_str(&process_directory(
+        "imported_mainnet_txns",
+        "imported_mainnet_txns",
+    ));
+    all_transactions_code.push_str(&process_directory(
+        "imported_testnet_txns",
+        "imported_testnet_txns",
+    ));
+    all_transactions_code.push_str(&process_directory("generated_txns", "generated_txns"));
+
+    // Write the generated Rust code to the output file
+    fs::write(dest_path, all_transactions_code).unwrap();
+}
+
+fn process_directory(dir_name: &str, module_name: &str) -> String {
     let mut transactions_code = String::new();
-    let json_dir = Path::new("json_transactions");
-    for entry in fs::read_dir(json_dir).unwrap() {
-        let entry = entry.unwrap();
+    let mut all_constants = String::new();
```

Comment:
What are these constants?

Reply:
this build.rs creates a rust file containing all the constans that look like this

```

                pub const IMPORTED_MAINNET_TXNS_MAINNET_V308783012: &[u8] = include_bytes!(concat!(env!("CARGO_MANIFEST_DIR"), "/json_transactions/imported_mainnet_txns/mainnet_v308783012.json"));
                pub const ALL_IMPORTED_MAINNET_TXNS: &[&[u8]] = &[IMPORTED_MAINNET_TXNS_MAINNET_V308783012,];
                pub const IMPORTED_TESTNET_TXNS_ACCOUNT_B_TRANSFER_TO_ACCOUNT_B: &[u8] = include_bytes!(concat!(env!("CARGO_MANIFEST_DIR"), "/json_transactions/imported_testnet_txns/account_b_transfer_to_account_b.json"));
                
                pub const IMPORTED_TESTNET_TXNS_ACCOUNT_A_GET_FUNDED_FROM_FAUCET: &[u8] = include_bytes!(concat!(env!("CARGO_MANIFEST_DIR"), "/json_transactions/imported_testnet_txns/account_a_get_funded_from_faucet.json"));
                
                pub const IMPORTED_TESTNET_TXNS_ACCOUNT_A_TRANSFER_TO_ACCOUNT_B: &[u8] = include_bytes!(concat!(env!("CARGO_MANIFEST_DIR"), "/json_transactions/imported_testnet_txns/account_a_transfer_to_account_b.json"));
                
                pub const IMPORTED_TESTNET_TXNS_TESTNET_V51268254: &[u8] = include_bytes!(concat!(env!("CARGO_MANIFEST_DIR"), "/json_transactions/imported_testnet_txns/testnet_v51268254.json"));
                
                pub const IMPORTED_TESTNET_TXNS_VALIDATOR_TRANSACTION: &[u8] = include_bytes!(concat!(env!("CARGO_MANIFEST_DIR"), "/json_transactions/imported_testnet_txns/validator_transaction.json"));
                pub const ALL_IMPORTED_TESTNET_TXNS: &[&[u8]] = &[IMPORTED_TESTNET_TXNS_ACCOUNT_B_TRANSFER_TO_ACCOUNT_B,IMPORTED_TESTNET_TXNS_ACCOUNT_A_GET_FUNDED_FROM_FAUCET,IMPORTED_TESTNET_TXNS_ACCOUNT_A_TRANSFER_TO_ACCOUNT_B,IMPORTED_TESTNET_TXNS_TESTNET_V51268254,IMPORTED_TESTNET_TXNS_VALIDATOR_TRANSACTION,];
                pub const GENERATED_TXNS_SIMPLE_USER_SCRIPT: &[u8] = include_bytes!(concat!(env!("CARGO_MANIFEST_DIR"), "/json_transactions/generated_txns/simple_user_script.json"));
                
                pub const GENERATED_TXNS_GENERATED_USER_SCRIPT_TRANSACTION: &[u8] = include_bytes!(concat!(env!("CARGO_MANIFEST_DIR"), "/json_transactions/generated_txns/generated_user_script_transaction.json"));
                pub const ALL_GENERATED_TXNS: &[&[u8]] = &[GENERATED_TXNS_SIMPLE_USER_SCRIPT,GENERATED_TXNS_GENERATED_USER_SCRIPT_TRANSACTION,];
                
                ```